### PR TITLE
ADR for state configuration

### PR DESCRIPTION
**Pull request by joeyg**
This PR includes a new ADR for configuration of iv-verify. While the PR includes recommendations please comment on this PR for discussion of the options.

### CODE REVIEW

**Comment by joeyg** [comment]
@YvetteWhiteUSDS @kategreenUSDS - for your comment.

File: adr/2024-09-states.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+#### Storage in a separate service
+
+This option allows for all configuration to be separated from the iv-verify service and put in it's own configuration service.
+
+
+#### Storage of configuration in file
+
+This option would see the configuration put in a json or yaml file that is bundled with the app code on deploy. The advantages of this solution is that there is version control for all configuration changes and it does not require standing up a database or separate service. The downside is that to make a configuration change a new deploy would have to be done.
+
+### Storage of state specific copy
```
Comment (kategreenUSDS) [inline adr/2024-09-states.md:34]: omg i didn't even think of this issue. so glad you did

File: adr/2024-09-states.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+In this option the URL for the ledger add screen states the same but when the user initiates the flow a cookie is set to indicate which flow the user is in. Both the client and server will have access to this cookie to navigate the user down the flow. The downside to this solution is that is not visibly clear which flow a user is unless the UI is modified to say so.
+
+## Decision
+
+The recommendations are as follows:
+
+### Storage
+
+Configuration should be stored in files for iv-verify. This requires the least upfront work and provides version control for configuration changes. iv-verify is not at the point where making configuration changes without a deploy is necessary.
```
Comment (kategreenUSDS) [inline adr/2024-09-states.md:64]: i concur on this point for the early stages. eventually we may want to move to something else, but for an MVP, this is where i landed as well

File: adr/2024-09-states.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+### Routing
+
+Routing should be handled through the URL. This option may require additional up front work as the URLs will need to be altered but it does create a clear system for indicating which configuration is being used. As iv-verify currently utalizes a high degree of static rendering which can be easier done on by URL.
+
+#### Copy storage
+
+Copy should be stored in the same file but with different keys. This is the easiest option to get up and running.
```
Comment (kategreenUSDS) [inline adr/2024-09-states.md:72]: i wonder if there's a way to do some magical routing with the keys to make retrieval less onerous. or if using the selected state from the url (or cookie should we go that route), we can use the separate file option. 

i wonder if there is a way to namespace the keys. managing the keys is onerous even with two languages (barely) and one state

File: adr/2024-09-states.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+The recommendations are as follows:
+
+### Storage
+
+Configuration should be stored in files for iv-verify. This requires the least upfront work and provides version control for configuration changes. iv-verify is not at the point where making configuration changes without a deploy is necessary.
+
+### Routing
+
+Routing should be handled through the URL. This option may require additional up front work as the URLs will need to be altered but it does create a clear system for indicating which configuration is being used. As iv-verify currently utalizes a high degree of static rendering which can be easier done on by URL.
```
Comment (kategreenUSDS) [inline adr/2024-09-states.md:68]: agree here as well, though a downside of this is that users could change the URL themselves or get themselves lost.

**Review by kategreenUSDS — APPROVED** [review]
2/3 you have my concurrence on.

On config files, I will likely add to this or create a new ADR for data structure for the config file (Julie and I were talking about what might go in the config today)

On routing, totally agree. 

On content storage, I'm feeling so overwhelmed by this, I need to digest it a bit and get back with you. 

Suggest we merge this with adding the ones we agree on and any consequences and fialize it. Then, we can move content storage to a separate ADR.

File: adr/2024-09-states.md (hunk 2 above)
Comment (tdooner) [reply adr/2024-09-states.md:64]: Here is [our configuration file for reference](https://github.com/DSACMS/iv-cbv-payroll/blob/main/app/config/site-config.yml) if it's helpful. We have a mix of checking in configuration as YAML and using environment variable interpolation.

File: adr/2024-09-states.md (hunk 3 above)
Comment (tdooner) [reply adr/2024-09-states.md:72]: We have a [`site_translation` helper](https://github.com/DSACMS/iv-cbv-payroll/blob/f3af04833520523a354644c3ef16098cf3332128/app/app/helpers/application_helper.rb#L8-L56) that overrides our `t` function by looking for a `.nyc` or `.ma` suffix for the given string, based on which site the user is viewing at the time. It falls back to a `.default` suffix if the site ID doesn't exist. ([e.g.](https://github.com/DSACMS/iv-cbv-payroll/blob/f3af04833520523a354644c3ef16098cf3332128/app/config/locales/en.yml#L35-L37)) It works pretty well - I recommend the pattern!

**Review by tdooner — APPROVED** [review]
Not to creep on y'all's work or anything but I saw a couple things in here that are similar to how we do it on CBV and wanted to share for inspiration.

**Review by tdooner — COMMENTED** [review]
Oops, didn't mean to click the "approve" button.

File: adr/2024-09-states.md (hunk 2 above)
Comment (kategreenUSDS) [reply adr/2024-09-states.md:64]: thank you thank you!

**Comment by kategreenUSDS** [comment]
@tdooner not at all creepin! appreciate your eyes and experience

**Pull request merged**